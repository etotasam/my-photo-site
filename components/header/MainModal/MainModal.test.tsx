import React from "react";
import { MainModal } from ".";
import { fireEvent, render, screen } from "@testing-library/react";
import { cleanup } from "@testing-library/react-hooks";

const nextRouer = jest.spyOn(require("next/router"), `useRouter`);
const push = jest.fn();
const query = { photo_label: "egypt" };
nextRouer.mockImplementation(() => {
  return { push, query };
});

const props = {
  locations: ["jordan", "egypt"],
  error: { message: "エラーです", name: "エラー" },
  photoLabelName: "jordan",
};

afterEach(() => {
  cleanup();
});

describe(`MainModal`, () => {
  // it(`propsで受け取るerrorの有無での表示検証 `, () => {
  //   const { rerender, queryByText, asFragment } = render(<MainModal {...props} />);
  //   expect(queryByText(/データ取得/)).toBeInTheDocument();
  //   expect(queryByText(/Jordan/)).toBeNull();
  //   expect(queryByText(/Egypt/)).toBeNull();
  //   expect(asFragment()).toMatchSnapshot();
  //   rerender(<MainModal locations={props.locations} />);
  //   expect(queryByText(/データ取得/)).toBeNull();
  //   expect(queryByText(/Jordan/)).toBeInTheDocument();
  //   expect(queryByText(/Egypt/)).toBeInTheDocument();
  //   expect(asFragment()).toMatchSnapshot();
  // });

  it(`router.pushのURLの検証`, async () => {
    const { queryByText } = render(<MainModal locations={props.locations} photoLabelName={props.photoLabelName} />);
    expect(queryByText(/Jordan/)).toBeInTheDocument();
    fireEvent.click(queryByText(/Jordan/) as HTMLElement);
    expect(push).toHaveBeenCalledWith(`/photo/jordan`);
  });
});
