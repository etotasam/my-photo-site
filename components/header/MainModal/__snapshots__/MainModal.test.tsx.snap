// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainModal propsで受け取るerrorの有無での表示検証  1`] = `
<DocumentFragment>
  <div
    class="t-modal-height bg-white w-full flex justify-center items-center"
    style="opacity: 0;"
  >
    <div>
      <p>
        データ取得に失敗しました 一度更新してください
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MainModal propsで受け取るerrorの有無での表示検証  2`] = `
<DocumentFragment>
  <div
    class="t-modal-height bg-white w-full flex justify-center items-center"
    style="opacity: 0;"
  >
    <div
      class="border border-gray-400 px-5 py-7 min-w-[200px]"
    >
      <ul>
        <li
          class="text-center pb-2 last-of-type:pb-0"
        >
          <a
            class="text-gray-500 cursor-pointer"
          >
            Jordan
          </a>
        </li>
        <li
          class="text-center pb-2 last-of-type:pb-0"
        >
          <a
            class="text-green-600"
          >
            Egypt
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
